native/src/components/DatePicker.tsx

```diff
@@ -0,0 +1,139 @@
+import { DateTime } from 'luxon'
+import React, { ReactElement, useEffect, useState } from 'react'
+import styled from 'styled-components/native'
+
+import { DATE_FORMAT } from 'shared/constants'
+
+import { CalendarTodayIcon } from '../assets'
+import Icon from './base/Icon'
+import IconButton from './base/IconButton'
+import Text from './base/Text'
+
+const DateContainer = styled.View`
+  width: fit-content;
+  position: relative;
+`
+const Input = styled.TextInput`
+  text-align: center;
+`
+const StyledInputWrapper = styled.View`
+  min-width: 316px;
+  height: 56px;
+  padding: 0 16px;
+  border-radius: 8px;
+  border-color: ${props => props.theme.colors.themeColorLight};
+  border-width: 3px;
+  border-style: solid;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+`
+const Wrapper = styled.View`
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+`
+const StyledIconButton = styled(IconButton)<{ $isModalOpen: boolean }>`
+  width: 40px;
+  height: 40px;
+  background-color: ${props =>
+    props.$isModalOpen ? props.theme.colors.themeColorLight : props.theme.colors.textDisabledColor};
+`
+const StyledTitle = styled.Text`
+  background-color: ${props => props.theme.colors.backgroundColor};
+  position: absolute;
+  top: -12px;
+  left: 12px;
+  right: auto;
+  padding: 2px 5px;
+  font-size: 12px;
+  font-weight: 400;
+  z-index: 1;
+`
+const StyledError = styled.Text`
+  font-size: 12px;
+  font-weight: bold;
+  color: ${props => props.theme.colors.invalidInput};
+`
+export type DatePickerProps = {
+  title: string
+  value: DateTime | null
+  setValue: (fromDate: DateTime | null) => void
+  error?: string
+  modalState: boolean
+  setModalState: React.Dispatch<React.SetStateAction<boolean>>
+}
+const DatePicker = ({ title, value, setValue, error, modalState, setModalState }: DatePickerProps): ReactElement => {
+  const formatDate = (date: DateTime | null, part: number) => date?.toFormat(DATE_FORMAT).split('/')[part]
+
+  const [inputDay, setInputDay] = useState(formatDate(value, 0))
+  const [inputMonth, setInputMonth] = useState(formatDate(value, 1))
+  const [inputYear, setInputYear] = useState(formatDate(value, 2))
```

Comment:
Date here needed to be formatted "dd/MM/yyyy" then split in to 3 parts for each (if I used date.day it wont be 02 for example I need it to be 2 digits).

Reply:
Oh, makes sense. Personally I'd like something like the following still better than some string magic. If the `DATE_FORMAT` is ever changed, it might lead to hard to catch errors. Like this, it is also immediatly clear what the output is.
```suggestion
  const [inputDay, setInputDay] = useState(value.toFormat('dd'))
  const [inputMonth, setInputMonth] = useState(value.toFormat('MM'))
  const [inputYear, setInputYear] = useState(value.toFormat('yy'))
```